elfi/methods/parameter_inference.py

```diff
@@ -941,54 +940,115 @@ def _report_batch(self, batch_index, params, distances):
             str += "{}{} at {}\n".format(fill, distances[i].item(), params[i])
         logger.debug(str)
 
-    def plot_state(self, **options):
-        """Plot the GP surface
-        
-        Currently supports only 2D cases.
+    def plot_acq_points(self):
+        """Plot the acquisition points.
         """
 
-        f = plt.gcf()
-        if len(f.axes) < 2:
-            f, _ = plt.subplots(1,2, figsize=(13,6), sharex='row', sharey='row')
-
         gp = self.target_model
+        n_params = gp.X.shape[1]
 
-        # Draw the GP surface
-        visin.draw_contour(gp.predict_mean,
-                           gp.bounds,
-                           self.parameter_names,
-                           title='GP target surface',
-                           points = gp.X,
-                           axes=f.axes[0], **options)
+        if n_params == 1:
+            print('Use plot_state() to visualise the acquisition points'
+                'in 1D.')
+            return
 
-        # Draw the latest acquisitions
-        if options.get('interactive'):
-            point = gp.X[-1, :]
-            if len(gp.X) > 1:
-                f.axes[1].scatter(*point, color='red')
+        fig, arr_ax = plt.subplots(nrows=n_params, ncols=n_params,
+            figsize=(10, 10))
+        fig.tight_layout(pad=2.0)
 
-        displays = [gp._gp]
+        for i in range(n_params):
+            # Plotting the pair-wise comparison.
+            for j in range(i + 1, n_params):
+                arr_ax[i, j].scatter(gp.X[:, i], gp.X[:, j])
+                arr_ax[i, j].set_xlabel(self.parameter_names[i])
+                arr_ax[i, j].set_ylabel(self.parameter_names[j])
+            # Plotting the marginals.
+            arr_ax[i, i].hist(gp.X[:, i], bins=20)
+            arr_ax[i, i].set_xlabel(self.parameter_names[i])
 
-        if options.get('interactive'):
-            from IPython import display
-            displays.insert(0, display.HTML(
-                    '<span><b>Iteration {}:</b> Acquired {} at {}</span>'.format(
-                        len(gp.Y), gp.Y[-1][0], point)))
-
-        # Update
-        visin._update_interactive(displays, options)
-
-        acq = lambda x : self.acquisition_method.evaluate(x, len(gp.X))
-        # Draw the acquisition surface
-        visin.draw_contour(acq,
-                           gp.bounds,
-                           self.parameter_names,
-                           title='Acquisition surface',
-                           points = None,
-                           axes=f.axes[1], **options)
-
-        if options.get('close'):
-            plt.close()
+    def plot_state(self, **options):
```

Comment:
Could you refactor this such that most of this stuff would be in the other modules dealing with visualization (`interactive.py` and `visualization.py`)? Ideally we would not need to import any plotting modules in `parameter_inference.py`.